doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+for every face is the following:
+`[image_id, label, conf, x_min, y_min, x_max, y_max]`;
+all the seven elements are floating point. For our further goals we need to take
+only results with `image_id > 0` because a negative value in this field
+indicates the end of detections; also we can cut detections by the `conf` field
+to avoid detector's mistakes; so the Mat parsing looks like this code:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp vec_ROI
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp fd_pp
+
+Some points to be mentioned about this kernel:
+
+- It takes in a `cv::Mat` of detections and a `cv::Mat` of the input frame; it
+gives out an array of ROI's where faces have been detected.
+
+- There is `cv::Mat` data parsing by the pointer on a float used.
+
+- Float numbers we've got from the Mat are between 0 and 1 and denote normalized
+coordinates; to get the real pixel coordinates we should multiply it by the
+image sizes respectively to the directions. To be fully accurate, the received
+numbers should be then rounded the right way and casted to `int` to construct
+integer Points; these operations have been wrapped in `toIntRounded()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp toInt
+
+- Rectangles are constructed by two Points --- the top-left and the bottom-right
+corners.
+
+- By far the most important thing here is solving an issue that sometimes
+detector spits out coordinates located outside of the image; if we pass such an
+ROI to be processed, errors of the landmarks detector will occure. The frame box
+`borders` is created and then intersected with the face rect (by `operator&()`)
+to handle these cases and save the ROI in `outFaces` which is for sure
+inside the frame.
+
+## Facial landmarks detector inference and post-processing {#gapi_fb_landm_detect}
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp ld_inf
+
+First of all, a few words about custom types: the `Contour` type is used for
+arrays of points which are naturally contours (ordered sets of points):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp cont
+
+The array with all landmarks is an array of points; syntactically it is a `Contour`
+though semantically is not a contour
+so I use another type `Landmarks` which syntactically is equal to `Contour`
+type to underline semantical differences:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp landm
+
+Post-processing of the Mat is required here too: there are 35 landmarks given by
+the detector for each face in a frame; the first 18 of them are facial elements
+(eyes, eyebrows, a nose, a mouth) and the last 17 --- a jaw contour. Landmarks
+are floating point values of coordinates normalized relatively to an input ROI
+(not the original frame). What is of the same importance, we need contours of
```

Comment:
`What is of the same importance` sounds really unnatural

Reply:
rewritten